### PR TITLE
Remove user from collection on deletion

### PR DESCRIPTION
**Pull request by thedesalizes**
This pull request updates the functionality of account deletion to include the removal of the user's entry in the users collection.

Summary of changes:
- Frontend
  - New function `removeFromUsersCollection()` in `UserFunctions.ts`
  - `DeleteAccountPageView.ts` calls this function during account deletion
- Backend
  - New route `user/removeFromUserCollection`
  - New controller function `removeFromUserCollection`

Any review and comments are appreciated :> Testing on my end was done from creation of account to deletion of account.

### CODE REVIEW

File: backend/user_service/controllers/userController.js [hunk 1]
```diff
@@ -24,6 +23,30 @@ const addToUserCollection = async (req, res) => {
     }
 };
 
+const removeFromUserCollection = async (req, res) => {
+    const { uid } = req.body;
+    if (!uid) {
+        return res.status(400).json({ error: "User ID (uid) is required." });
```
Comment (ekweirui) [inline backend/user_service/controllers/userController.js]: Change the field name from `error` to `message` to be in sync with how error messages are handled in `callFunction` in libs/util.ts.

File: backend/user_service/controllers/userController.js [hunk 2]
```diff
@@ -24,6 +23,30 @@ const addToUserCollection = async (req, res) => {
...
+    }
+
+    try {
+        // Attempt to delete the document with the provided UID in the "users" collection
+        const userRef = db.collection("users").doc(uid);
+        const userDoc = await userRef.get();
+
+        if (!userDoc.exists) {
+            return res.status(404).json({ error: "User not found." });
```
Comment (ekweirui) [inline backend/user_service/controllers/userController.js]: Similar point to above.

File: backend/user_service/controllers/userController.js [hunk 3]
```diff
@@ -24,6 +23,30 @@ const addToUserCollection = async (req, res) => {
...
+        }
+
+        await userRef.delete();
+
+        return res.status(200).json({ message: "User successfully removed from the collection." });
+    } catch (error) {
+        console.error("Error removing user:", error);
+        return res.status(500).json({ error: "Failed to remove user from the collection." });
```
Comment (ekweirui) [inline backend/user_service/controllers/userController.js]: Similar point to above.

File: frontend/src/services/UserFunctions.ts [hunk 4]
```diff
@@ -15,6 +15,15 @@ export async function addToUserCollection(userCredential: UserCredential, userna
     return res;
 }
 
+export async function removeUserFromCollection(): Promise<SuccessObject> {
+    const uid = sessionStorage.getItem("uid");
+    const userData = {
+        uid
```
Comment (ekweirui) [inline frontend/src/services/UserFunctions.ts]: Small issue but `uid: uid` might be clearer

File: frontend/src/services/UserFunctions.ts [hunk 5]
```diff
@@ -15,6 +15,15 @@ export async function addToUserCollection(userCredential: UserCredential, userna
...
+    }
+    const res = await callFunction(HTTP_SERVICE_USER, "user/removeFromUserCollection", "POST", userData);
+    return res
```
Comment (ekweirui) [inline frontend/src/services/UserFunctions.ts]: Missing semi colon.

**Review by ekweirui — COMMENTED** [review]
LGTM overall other than minor adjustments to be made. 👍

File: backend/user_service/controllers/userController.js (hunk 1 above)
Comment (thedesalizes) [reply backend/user_service/controllers/userController.js]: gdi i keep forgetting if its error or message, thanks for pointing it out

**Review by ekweirui — APPROVED** [review]
LGTM 👍 Thank you for the quick changes!

**Pull request merged**